allennlp/models/archival.py

```diff
@@ -190,18 +192,22 @@ def load_archive(archive_file: str,
 
     # Check for supplemental files in archive
     fta_filename = os.path.join(serialization_dir, _FTA_NAME)
-    if os.path.exists(fta_filename):
+    if not loading_dir and os.path.exists(fta_filename):
```

Comment:
is this the right logic? imagine that I have a model.tar.gz, and then I untar it somewhere and then point to that location. in that case we'd have `loading_dir == True` but we'd still want to grab the files from the "archive" (which is a directory)

Reply:
Good point. It does seem possible that some users are relying on this. I've changed the behavior again to act as it did before if the files are present, but to load their serialization directory equivalents otherwise.

(I also considered `touch`ing an `ARCHIVE_MARKER` or `SERIALIZATION_DIR_MARKER` that we could check explicitly, but it seem desirable to have this work for existing dirs and archives.)